Swiften/MIX/MIXImpl.cpp

```diff
@@ -94,4 +94,95 @@ void MIXImpl::updatePreferences(Form::ref form) {
     request->send();
 }
 
+void MIXImpl::registerNick(const std::string& nick) {
+    auto registerNickPayload = std::make_shared<MIXRegisterNick>();
+    registerNickPayload->setNick(nick);
+    auto request = std::make_shared<GenericRequest<MIXRegisterNick> >(IQ::Set, channelJID_, registerNickPayload, iqRouter_);
+    request->onResponse.connect(boost::bind(&MIXImpl::handleNickAssigned, this, _1, _2));
+    request->send();
+}
+
+void MIXImpl::handleNickAssigned(MIXRegisterNick::ref payload, ErrorPayload::ref error) {
+    std::string nick;
+    if (payload) {
+        nick = payload->getNick();
+    }
+    onNickResponse(nick, error);
+}
+
+void MIXImpl::setNick(const std::string& nick) {
+    auto setNickPayload = std::make_shared<MIXSetNick>();
+    setNickPayload->setNick(nick);
+    auto request = std::make_shared<GenericRequest<MIXSetNick> >(IQ::Set, channelJID_, setNickPayload, iqRouter_);
+    request->onResponse.connect(boost::bind(&MIXImpl::handleNickUpdated, this, _1, _2));
+    request->send();
+}
+
+void MIXImpl::handleNickUpdated(MIXSetNick::ref payload, ErrorPayload::ref error) {
+    std::string nick;
+    if (payload) {
+        nick = payload->getNick();
+    }
+    onNickResponse(nick, error);
+}
+
+void MIXImpl::getSupportedFeatures() {
+    auto discoInfo = std::make_shared<DiscoInfo>();
+    discoInfo->setNode(std::string("mix"));
+    auto request = std::make_shared<GenericRequest<DiscoInfo> >(IQ::Get, channelJID_, discoInfo, iqRouter_);
+    request->onResponse.connect(boost::bind(&MIXImpl::handleFeaturesReceived, this, _1, _2));
+    request->send();
+}
+
+void MIXImpl::handleFeaturesReceived(DiscoInfo::ref payload, ErrorPayload::ref /*error*/) {
+    if (payload) {
+        assert(payload->getNode() == std::string("mix"));
+        supportedFeatures_ = payload;
+    }
+}
+
+void MIXImpl::getSupportedNodes() {
+    auto discoItems = std::make_shared<DiscoItems>();
+    discoItems->setNode(std::string("mix"));
+    auto request = std::make_shared<GenericRequest<DiscoItems> >(IQ::Get, channelJID_, discoItems, iqRouter_);
+    request->onResponse.connect(boost::bind(&MIXImpl::handleSupportedNodesReceived, this, _1, _2));
+    request->send();
+}
+
+void MIXImpl::handleSupportedNodesReceived(DiscoItems::ref payload, ErrorPayload::ref /*error*/) {
+    if (payload) {
+        assert(payload->getNode() == std::string("mix"));
```

Comment:
Maybe use `SWIFT_LOG_ASSERT` instead, which only writes a message and doesn't exit the program on condition. Same goes for the other `assert` usages in `MIXImpl.cpp`.